portal_tables/unify_grant_tables.py

```diff
@@ -0,0 +1,145 @@
+"""Combine Synapse projects that are separated 
+by grant into the CCKP admin project table.
+
+This script assumes the folders for PublicationView,
+DatasetView, and ToolView, respectively, remain 
+consistent in the CCKP admin project table.
+
+author: victor.baham
+"""
+
+# command line arguments: 1. Portal - Grants Merged Synapse ID 2. CCKP Admin Project Synapse ID
+# test value for argument 1: syn42801895
+# test value for argument 2: syn35629947 
+
+import argparse
+import synapseclient
+from synapseclient import Project, File, Folder
+from synapseclient import Schema, Column, Table, Row, RowSet, as_table_columns
+import itertools
+import pandas as pd
+
+def get_args():
+    #Retrieve Synapse table IDs from command line
+    parser = argparse.ArgumentParser(
+        description="Merge publications, datasets, and tools from Grants Merged table into CCKP admin table")
+    parser.add_argument("-p1", "--project1",
+                        type=str, required=True,
+                        help="Portal - Grants Merged Synapse ID.")
+    parser.add_argument("-p2", "--project2",
+                        type=str, required=True,
+                        help=("CCKP - MC2 Admin Synapse ID"))
+    parser.add_argument("--dryrun", action="store_true")
+    return parser.parse_args()
+
+def getManifestsFromGrant(syn,theParent):
+    """This function examines the GrantView table
+    and returns a list of all Synapse table IDs
+    separated by manifest type (publications, datasets,
+    tools) for each grant"""
+
+    # Access each grant project by accessing 'grantId'
+    # column in grantView manifest
+    # synId for grantView is hardcoded in and
+    # will need to be changed for future use
+    df = syn.tableQuery(f"select {'grantId'} from {theParent}").asDataFrame()
+    gIds = df['grantId'].tolist()
+
+    # for each grant, get publications,
+    # datasets, and tools 
+
+    # value inside dictionary will reset upon each
+    # loop iteration, so appending to this list 
+    # maintains each one
+    mTypes = []
+    mIds = []
+    for gId in gIds:
+        grantX = syn.get(gId)
+        # By accessing the grants, it is possible
+        # to access their 'children'
+
+        # extract manifest type and table id from generator
+        mTypes.append([m['name'] for m in syn.getChildren(grantX, includeTypes=['table'])])
```

Comment:
Generally, it's almost always better to use `.get()` rather than the square notation when accessing a dictionary:

```suggestion
        mTypes.append([m.get('name') for m in syn.getChildren(grantX, includeTypes=['table'])])
```

With this approach, if the key isn't found, then it'll return `None`.  Using the square brackets will result in the `KeyError` exception if the key isn't found.

That being said, here it is safe to assume that `name` will always be found, so you don't need to use `get()` here.  Just letting you know for future scripts :smile: